### DISCUSSION TITLE
Experimenting with boundary conditions

### DISCUSSION THREAD

**Discussion by kerim371** [slimgroup/JUDI.jl#209] [Q&A]
Hi,

I feel like default boundary condition requires to greatly expand computational grid so it absorbs waves.
I would like to play around this.
Especially I wanted to compare Absorbing Boundary Conditions (ABC) and Perfectly Matched Layers (PML).
Could you please guide me where it is implemented in JUDI and how can I start start modifying BC?
Also for now JUDI uses ABC or PML?

**Comment by mloubout** [comment]
The current boundary condition is a diffusion mechanism that translate to solving the equation

$$
m \frac{d^2 u}{dt^2} - \Delta u + \eta \frac{du}{dt} = 0
$$

where $\eta$. is zero in the physical domain and increasing in the boundary layer to progressively damp and diffuse the waves in it. 

The damping profile is implemented [here](https://github.com/slimgroup/JUDI.jl/blob/83752f147e175608d98b2ffe2d4778d58281c1f1/src/pysource/models.py#L65)

and the pde in [kernels.py](https://github.com/slimgroup/JUDI.jl/blob/master/src/pysource/kernels.py)

If you want to look at something fancier like PMLs you might wanna look at this devito tutorials

https://github.com/devitocodes/devito/tree/master/examples/seismic/abc_methods

**Comment by kerim371** [comment]
@mloubout according to the Devito examples, PML expects `ndims*2` (or probably `ndims` if they are pairwisely equal) damping fields. 
Thus I guess we have to overwrite `Model.damp` from `Function` type to `Tuple(Function)`.
Then I probably have to reimplement [damp_op](https://github.com/slimgroup/JUDI.jl/blob/83752f147e175608d98b2ffe2d4778d58281c1f1/src/pysource/models.py#L102C1-L104) method so it return `Tuple(Operator)` (one operator for damp field) and call each of them within [initialize_damp()](https://github.com/slimgroup/JUDI.jl/blob/83752f147e175608d98b2ffe2d4778d58281c1f1/src/pysource/models.py#L127C1-L128C18) method.
Finally modify kernel for each of type of equation.

May this work?

**Comment by kerim371** [comment]
Draft PR #210
